### PR TITLE
Added new features to bucket replication rules panel

### PR DESCRIPTION
**Pull request by bexsoft**
## What does this do?

- Added Edit option for replication
- Sorted by priority in replication list
- Fixed an issue where storage class was not saved for replication rule

**NOTE:** Please test add & edit rules roughly as both modules were changed

## How does it look?
Screen Shot 2022-01-18 at 21 42 59

Screen Shot 2022-01-19 at 19 17 44
Screen Shot 2022-01-19 at 19 17 33

Signed-off-by: Benjamin Perez <benjamin@bexsoft.net>

### CODE REVIEW

**Review by dvaldivia — APPROVED** [review]
LGTM.

**Review by dvaldivia — APPROVED** [review]

File: models/multi_bucket_replication_edit.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	// replicate delete markers
+	ReplicateDeleteMarkers bool `json:"replicateDeleteMarkers,omitempty"`
+
+	// replicate deletes
+	ReplicateDeletes bool `json:"replicateDeletes,omitempty"`
+
+	// replicate metadata
+	ReplicateMetadata bool `json:"replicateMetadata,omitempty"`
+
```
Comment (poornas) [inline models/multi_bucket_replication_edit.go:54]: @bexsoft , model should also have a flag for replicating existing objects

**Review by poornas — CHANGES_REQUESTED** [review]

File: models/multi_bucket_replication_edit.go (hunk 1 above)
Comment (bexsoft) [reply models/multi_bucket_replication_edit.go:57]: Fixed, added support to replicate existing objets & added selectors to form

**Comment by dvaldivia** [comment]
@poornas PTAL

**Comment by bexsoft** [comment]
Added selector for metadata replication. Regarding removing the validation for not deleting the last rule, minio-go sdk still contains this restriction. Will keep this behavior for now
Screen Shot 2022-01-19 at 23 25 44
Screen Shot 2022-01-19 at 23 25 37
Screen Shot 2022-01-19 at 23 25 28
Screen Shot 2022-01-19 at 23 20 57
Screen Shot 2022-01-19 at 23 19 05

**Review by poornas — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**